mongo/datadog_checks/mongo/collectors/replica.py

```diff
@@ -0,0 +1,104 @@
+import time
+
+from six.moves.urllib.parse import urlsplit
+
+from datadog_checks.mongo.collectors.base import MongoCollector
+from datadog_checks.mongo.common import REPLSET_MEMBER_STATES, SOURCE_TYPE_NAME, get_state_name
+
+try:
+    import datadog_agent
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+
+class ReplicaCollector(MongoCollector):
+    """Collect replica set metrics by running the replSetGetStatus command. Also keep track of the previous node state
+    in order to submit events on any status change.
+    """
+
+    def __init__(self, check, tags, last_state):
+        super(ReplicaCollector, self).__init__(check, "admin", tags)
+        # Members' last replica set states
```

Comment:
(?) I think we're keeping track of the state of a _given_ member, not several members?

```suggestion
        # Member's last replica set state
```